libyul/Dialect.h

```diff
@@ -49,7 +49,7 @@ struct BuiltinFunction
 	/// If true, this is the msize instruction.
 	bool isMSize = false;
 	/// If set, same length as the arguments, if true at index i, the i'th argument has to be a literal which means it can't be moved to variables.
-	std::optional<std::vector<bool>> literalArguments;
+	std::optional<std::vector<std::optional<LiteralKind>>> literalArguments;
```

Comment:
Alternative is creating yet another enum based on LiteralKind which has a none value, but I think that is worse.

We could consider removing the outside optional wrapper though.

Reply:
What about ``bool hasLiteralArgument = contains_if(literalArguments, [](auto const& _arg) { return _arg.has_value(); });`` or sth like that?

Reply:
We'd need a helper in either case because it used/needed in multiple places. Can we perhaps add it onto the `BuiltinFunction` struct?

Reply:
Another major refactoring option is changing `parameters` to take a combined type which includes literals :)